sphinx/ext/inheritance_diagram.py

```diff
@@ -76,14 +76,18 @@ def try_import(objname):
         __import__(objname)
         return sys.modules.get(objname)  # type: ignore
     except ImportError:
-        modname, attrname = module_sig_re.match(objname).groups()  # type: ignore
-        if modname is None:
-            return None
         try:
-            __import__(modname)
-            return getattr(sys.modules.get(modname), attrname, None)
-        except ImportError:
+            modname, attrname = module_sig_re.match(objname).groups()  # type: ignore
+        except AttributeError:
```

Comment:
How about this? It would be better to check the matched or not before calling `group()` to me.
```
matched = module_sig_re.match(objname)  # type: ignore
if not matched:
    return None

modname, attrname = matched.groups()
...
```

Reply:
Yeh, really better :)  And I will provide the test.